### ISSUE TITLE
non-default Debian `umask` issue causes insufficient permissions on `webroot` tree

### ISSUE THREAD

**Issue by corbolais** [pi-hole/pi-hole#3175]
**TL;DR**: `umask` issue. #2413 (curtesy of @DrGeoff) tipped me off: umask of 0077 lead to inaccessible `/var/www/html/admin`, `/var/www/html/pihole` permissions were fine, though!

**Solution**: Fix permissions of `/var/www/html/admin/*`.

Offending line in `automated install/basic_install.sh`:
https://github.com/pi-hole/pi-hole/blob/9e490775ff3b20f378acc9db7cec2ae6023fff7f/automated%20install/basic-install.sh#L1894

This happened when deploying pihole on a Debian machine using `sslh` (both openvpn an httpd use port 80) and having

> server.bind = "127.0.0.1"
> server.port = 8080

in `lighttpd.conf`. Additionally, `pihole` is listening in on `tun0`, ip `10.x.x.1#53` with `pihole-FTL` equally listening in on `tun0`, ip `10.x.x.1` . Requests get forwarded to `unbound`, listening on `127.0.0.1#5353`. A completely separate `bind/named` is listening on `127.0.0.1#53`.

Replacing 
> $serverName = htmlspecialchars($_SERVER["HTTP_HOST"]);

by

> $serverName = htmlspecialchars($_SERVER["SERVER_ADDR"]);

in `/var/www/html/pihole/index.php` as per @unitpas resolved this issue.

cheers

_Originally posted by @corbolais in https://github.com/pi-hole/pi-hole/issues/2195#issuecomment-583777187_
